tests/integration_tests/instances.py

```diff
@@ -81,7 +83,21 @@ def restart(self):
         This wraps pycloudlib's `BaseInstance.restart`
         """
         log.info("Restarting instance and waiting for boot")
-        self.instance.restart()
+        iter = 32
```

Comment:
Looking at this again with fresh eyes, I have a couple concerns:
1. The restart wait times in pycloudlib are already huge. If there's something preventing us from connecting, this loop will cause us to wait about 6 hours for a single test.
2. I've seen issues (particulary in Azure) where we get an exception back from the cloud API when issuing the restart. If it still results in a restart, we could wind up restarting the same instance multiple times.

I don't have a great solution, because the status quo is also not good.

I'm wondering if it would make sense to add some smarts in pycloudlib to wait a shorter amount of time, and if we still haven't rebooted (i.e., can still connect with same boot id), do a force stop and start of the instance.